Update information about the `places` collection

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request updates the documented information about the `places` collection. This includes:
- Update old, dead URLs to live equivalents.
- Re-order fields to mirror the [collection schema].
- Add additional fields to match the [collection schema].
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

I was working on how this data is imported in https://github.com/cisagov/cyhy-core/pull/83 and noticed that not all fields were represented in this data dictionary. I decided to do some general cleanup when I went to add the missing fields.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

[collection schema]: https://github.com/cisagov/cyhy-core/blob/9d867f753136747e709480c770cc7021c9253849/cyhy/db/database.py#L1672-L1689

mcdonnnj committed:
Update links for the places collection

Update old, dead URLs to live alternatives. Replace "GNIS ID" with
"Feature ID" to match the dataset. Add a URL to download the data
we're specifically using.

mcdonnnj committed:
Rearrange places collection field order

This reflects the order in the schema definition found in the
cisagov/cyhy-core project. Please see:
https://github.com/cisagov/cyhy-core/blob/9d867f753136747e709480c770cc7021c9253849/cyhy/db/database.py#L1672-L1689

mcdonnnj committed:
Add additional fields for a places document

These fields are defined in the collection schema and are populated for
some data sources.

dav3r reviewed (approved):
Thanks for the cleanup! 🥇

jsf9k commented on NCATS_Data_Dictionary.md:
```diff
@@ -282,28 +282,36 @@ notification refers to
+- `country` [string]: Two-letter abbreviation of the country where this
+  place is
+- `country_name` [string]: Full name of the country where this place is
+- `latitude_dms` [string]: Latitude of this place, in degrees, minutes,
+  and seconds
+- `longitude_dms` [string]: Longitude of this place, in degrees,
+  minutes, and seconds
```
Should we specify the format of the DMS string?  I've seen it done a few different ways, e.g., `DDDMMSS`, `DDD MM SS`.

mcdonnnj replied:
Please let me know if https://github.com/cisagov/ncats-data-dictionary/pull/43/commits/c0b7a4c67e6ecd8fd4528a83e3e8d944ecaae365 works for you. I also added https://github.com/cisagov/ncats-data-dictionary/pull/43/commits/857a112a278a0a90d5ebe5dfc0eff6caabad92f8 as a more general catch-all for the formats of this collection's fields.

jsf9k replied:
Perfect!

mcdonnnj committed:
Add information on the data formats for the `places` collection

Add a sentence directing readers to the GNIS document that details the
file formats for the source data used for this collection. Since we do
almost no data manipulation it is the best explanation for what the
data looks like.

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

mcdonnnj committed:
Add some format information about two `places` fields

Add a simple format description for the `latitude_dms` and
`longitude_dms` fields.

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>
